### PR TITLE
Jormungandr: fix reverse isochrons

### LINKED ISSUE #7: journeys entry point: "from" not required in doc but seems mandatory in API
If I try this call: https://api.sncf.com/v1/coverage/sncf/journeys?to=stop_area:OCE:SA:87686006&datetime=20150622T1000, I get this result:

```
{"message":"from argument is required"}
```

But the [doc](https://data.sncf.com/api/documentation#title.1.6.4) says it's not required.
Is the doc out of date or me missing something? :)

Thanks for your lights!

### PR DESCRIPTION
**Pull request by antoine-de**
isochrons were working only with a 'from'argument, they now also works with only a 'to' argument.

fix http://jira.canaltp.fr/browse/NAVITIAII-1774

@yohanboniface, it should fix your problem, thanks for the bug report!

fixes https://github.com/SNCFdevelopers/API-trains-sncf/issues/7

### CODE REVIEW

File: source/kraken/worker.cpp [hunk 1]
```diff
@@ -567,20 +577,27 @@ pbnavitia::Response Worker::journeys(const pbnavitia::JourneysRequest &request,
...
-        case pbnavitia::NMPLANNER:
-            return routing::make_nm_response(*planner, origins, destinations, datetimes[0],
+    case pbnavitia::ISOCHRONE: {
+        type::EntryPoint ep;
+        if (! origins.empty()) {
+            ep = origins[0];
+        } else {
+            ep = destinations[0];
+        }
+        return navitia::routing::make_isochrone(*planner, ep, request.datetimes(0),
```
Comment (kinnou02) [inline source/kraken/worker.cpp:601]: Don't you need to specify if `ep` is the origin or the destination?

File: source/jormungandr/tests/isochrone_tests.py [hunk 2]
```diff
@@ -43,41 +43,61 @@ def test_from_isochrone_coord(self):
...
+        self.query(query)
+
+    def test_reverse_isochrone_coord(self):
+        q = "v1/coverage/basic_routing_test/journeys?max_duration=25500&datetime=20120614T080000"
+        normal_response = self.query(q + "&from=A")
+        assert len(normal_response["journeys"]) == 2
+
+        # the reverse isochrone should also find 2 journeys
+        normal_response = self.query(q + "&to=A")
+        assert len(normal_response["journeys"]) == 2
```
Comment (TeXitoi) [inline source/jormungandr/tests/isochrone_tests.py]: Please add a `\n` at the end of the file.

File: source/jormungandr/tests/isochrone_tests.py [hunk 3]
```diff
@@ -43,41 +43,61 @@ def test_from_isochrone_coord(self):
...
         #not to use the jormungandr database
         query = "v1/coverage/basic_routing_test/journeys?from={}&datetime={}&datetime_represents=arrival"
         query = query.format("stopA", "20120614T080000")
-        self.query(query, display=False)
+        self.query(query)
+
+    def test_reverse_isochrone_coord(self):
+        q = "v1/coverage/basic_routing_test/journeys?max_duration=25500&datetime=20120614T080000"
+        normal_response = self.query(q + "&from=A")
+        assert len(normal_response["journeys"]) == 2
```
Comment (TeXitoi) [inline source/jormungandr/tests/isochrone_tests.py:83]: We must test the datetimes somewhere.

**Comment by TeXitoi** [comment]
The direction is fixed by the from/to param in isochrone: from -> clockwise, to -> non-clockwise. Jormungandr should do something about that and kraken or raptor should check the coherence of these params.

**Comment by antoine-de** [comment]
you're right I added a test on this

File: source/jormungandr/tests/isochrone_tests.py (hunk 3 above)
Comment (antoine-de) [inline source/jormungandr/tests/isochrone_tests.py:83]: it is tested in routing_api_test

File: source/jormungandr/tests/isochrone_tests.py [hunk 4]
```diff
@@ -62,22 +62,37 @@ def test_stop_point_isochrone_coord(self):
...
+        assert error_code == 404
+        assert normal_response['error']['message'] == 'reverse isochrone works only for anti-clockwise request'
+
+
+    def test_isochrone_non_clockwise(self):
+        q = "v1/coverage/basic_routing_test/journeys?datetime=20120614T080000&from=A&datetime_represents=arrival"
+        normal_response, error_code = self.query_no_assert(q)
+
+        assert error_code == 404
+        assert normal_response['error']['message'] == 'isochrone works only for clockwise request'
```
Comment (TeXitoi) [inline source/jormungandr/tests/isochrone_tests.py]: linefeed

**Comment by TeXitoi** [comment]
OK for me

File: source/jormungandr/tests/isochrone_tests.py (hunk 4 above)
Comment (antoine-de) [inline source/jormungandr/tests/isochrone_tests.py]: done

**Comment by kinnou02** [comment]
Fine, but you need to update the documentation of isochrone

**Comment by antoine-de** [comment]
:fu:

**Comment by l-vincent-l** [comment]
@stifoon manage your team!

**Pull request merged**